src/FocusApp.Client/Views/Social/ProfilePage.cs

```diff
@@ -0,0 +1,506 @@
+                    Opacity = 0.5
+                }
+                .Row(PageRow.MembershipDate)
+                .ColumnSpan(typeof(PageColumn).GetEnumNames().Length)
+                .Bottom(),
+            }
+        };
+    }
+
+    // Refresh bindings on page load in case of profile edit / navigation to settings page
+    protected override async void OnAppearing()
+    {
+        ByteArrayToImageSourceConverter byteArrayConverter = new ByteArrayToImageSourceConverter();
+        _profilePicture.ImageSource = byteArrayConverter.ConvertFrom(_authenticationService.CurrentUser?.ProfilePicture);
+        _userName.Text = _authenticationService.CurrentUser?.UserName;
+        _pronouns.Text = $"Pronouns: {_authenticationService.CurrentUser?.Pronouns}";
+        _email.Text = $"Friend Id: #{_authenticationService.CurrentUser?.Email}";
+
+        _selectedPet.Source = byteArrayConverter.ConvertFrom(_authenticationService.CurrentUser?.SelectedPet?.Image);
+        _selectedIsland.Source = byteArrayConverter.ConvertFrom(_authenticationService.CurrentUser?.SelectedIsland?.Image);
+
+        if (_authenticationService.CurrentUser?.SelectedDecor != null)
+            _selectedDecor.Source = byteArrayConverter.ConvertFrom(_authenticationService.CurrentUser?.SelectedDecor?.Image);
+
+        if (_authenticationService.CurrentUser?.SelectedBadge != null)
+            _selectedBadge.Source = byteArrayConverter.ConvertFrom(_authenticationService.CurrentUser?.SelectedBadge?.Image);
+    }
+
+    private void CreateUserDataElements()
+    {
+        // Set bindable properties with images
+        _profilePicture = new AvatarView
+        {
+            CornerRadius = 60,
+            HeightRequest = 120,
+            WidthRequest = 120
+        }
+        .Bind(AvatarView.ImageSourceProperty,
+              "ProfilePicture",
+              converter: new ByteArrayToImageSourceConverter());
+        _profilePicture.BindingContext = _authenticationService.CurrentUser;
+
+        _userName = new Label
+        {
+            Text = $"{_authenticationService.CurrentUser?.UserName}",
+            FontSize = 18
+        };
+
+        _pronouns = new Label
+        {
+            Text = $"Pronouns: {_authenticationService.CurrentUser?.Pronouns}",
+            FontSize = 15
+        };
+
+        _email = new Label
+        {
+            Text = $"Friend Id: #{_authenticationService.CurrentUser?.Email}",
+            FontSize = 15
+        };
+    }
+
+    private void CreateSelectedItemElements()
+    {
+        _selectedPet = new ImageButton
+        {
+            HeightRequest = 110,
+            WidthRequest = 110
+        }
+        .Bind(ImageButton.SourceProperty, "Image", converter: new ByteArrayToImageSourceConverter())
+        .Invoke(button => button.Released += (sender, eventArgs) =>
+            SelectedPetClicked(sender, eventArgs));
+        _selectedPet.BindingContext = _authenticationService.CurrentUser?.SelectedPet;
```

Comment:
Although this does work, the CurrentUser properties aren't setup for binding/property changed logic, but the _authenticationService.Selected____ properties are. It makes me nervous to use the CurrentUser.Selected____ here for this reason. 

For consistency with the TimerPage & and for safety (especially safety with the inconsistency of Maui images), could we use _authenticationService.Selected____ instead of _authenticationService.CurrentUser.Selected___?

Reply:
Note that part of this is because I had a lot of issues when trying to bind to the CurrentUser properties when working on the IslandDisplayView (probably because of the navigation but still)